### PR TITLE
feat: add persistence to analytics module #584

### PR DESCRIPTION
**Pull request by gawicks**

### CODE REVIEW

File: pkg/storage/analytics.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	MemAlloc             int       `json:"mem_alloc"`
+	MemTotalAlloc        int       `json:"mem_total_alloc"`
+	MemSys               int       `json:"mem_sys"`
+	MemNumGC             int       `json:"mem_num_gc"`
+	BadgerMain           int       `json:"badger_main"`
+	BadgerTrees          int       `json:"badger_trees"`
+	BadgerDicts          int       `json:"badger_dicts"`
+	BadgerDimensions     int       `json:"badger_dimensions"`
+	BadgerSegments       int       `json:"badger_segments"`
+	ControllerIndex      int       `json:"controller_index" kind:"cumulative"`
```
Comment (gawicks) [inline pkg/storage/analytics.go:28]: wasn't really sure what to name this key. I've struck with kind

File: pkg/storage/analytics.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	ControllerDiff       int       `json:"controller_diff" kind:"cumulative"`
+	ControllerIngest     int       `json:"controller_ingest" kind:"cumulative"`
+	ControllerRender     int       `json:"controller_render" kind:"cumulative"`
+	SpyRbspy             int       `json:"spy_rbspy" kind:"cumulative"`
+	SpyPyspy             int       `json:"spy_pyspy" kind:"cumulative"`
+	SpyGospy             int       `json:"spy_gospy" kind:"cumulative"`
+	SpyEbpfspy           int       `json:"spy_ebpfspy" kind:"cumulative"`
+	SpyPhpspy            int       `json:"spy_phpspy" kind:"cumulative"`
+	SpyDotnetspy         int       `json:"spy_dotnetspy" kind:"cumulative"`
+	SpyJavaspy           int       `json:"spy_javaspy" kind:"cumulative"`
```
Comment (gawicks) [inline pkg/storage/analytics.go:39]: possibly some of these other metrices are also cumulative?

File: pkg/storage/analytics.go (hunk 1 above)
Comment (petethepig) [reply pkg/storage/analytics.go:28]: Yeah, this is good, and I like this approach of using struct tags.

File: pkg/storage/analytics.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		err := txn.SetEntry(entry)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (kolesnikovae) [inline pkg/storage/analytics.go:61]:
```suggestion
	return s.main.Update(func(txn *badger.Txn) error {
		return txn.SetEntry(badger.NewEntry([]byte(analyticsKey), v))
	})
```

File: pkg/storage/analytics.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+			return err
+		}
+		err = v.Value(func(val []byte) error {
+			err = json.Unmarshal(val, a)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		return nil
```
Comment (kolesnikovae) [inline pkg/storage/analytics.go:78]:
```suggestion
		return v.Value(func(val []byte) error {
			return json.Unmarshal(val, &a)
		})
```

Comment (gawicks) [reply pkg/storage/analytics.go]: thanks @kolesnikovae

**Pull request merged**